gui/wxpython/core/settings.py

```diff
@@ -25,18 +25,51 @@
 import sys
 import copy
 import wx
+import json
 
 from core import globalvar
 from core.gcmd import GException, GError
 from core.utils import GetSettingsPath, PathJoin, rgb2str
 
 
+class SettingsJSONEncoder(json.JSONEncoder):
+    """Custom JSON encoder"""
+    def default(self, obj):
+        """Encode not automatically serializable objects"""
+        if isinstance(obj, (wx.FontFamily, wx.FontStyle, wx.FontWeight)):
+            return int(obj)
```

Comment:
Since the int in this case does not make sense outside of wx, please add info on what would be the alternative here in case we want to make it more general/wx-independ in the future.